app/models/spree/product.rb

```diff
@@ -260,6 +261,11 @@ def import_date
       variants.map(&:import_date).compact.max
     end
 
+    def destroy(deleted_by: nil)
+      update_attribute(:deleted_by, deleted_by) if deleted_by.present?
```

Comment:
I am thinking we shouldn't have a default for `deleted_by` to enforce recording a user when deleting a product. But I am pretty sure it would blow up in various unexpected places.